torchgeo/models/copernicusfm.py

```diff
@@ -0,0 +1,724 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# https://github.com/zhu-xlab/Copernicus-FM
+
+"""Copernicus Foundation Model (Copernicus-FM)."""
+
+import collections.abc
+import math
+import os
+from itertools import repeat
+from typing import Any
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.nn.init as init
+from einops import rearrange
+from functorch import vmap
+from timm.models.vision_transformer import Block
+from torch import Tensor
+from torchvision.datasets.utils import download_url
+from torchvision.models._api import Weights, WeightsEnum
+
+
+def to_2tuple(x: Any) -> tuple:
+    """Borrowed from https://github.com/bwconrad/flexivit.
+    """
+    if isinstance(x, collections.abc.Iterable) and not isinstance(x, str):
+        return tuple(x)
+    return tuple(repeat(x, 2))
+
+
+def resize_abs_pos_embed(
+    pos_embed: torch.Tensor,
+    new_size: tuple[int, int],
+    old_size: int | tuple[int, int] | None = None,
+    num_prefix_tokens: int = 1,
+    interpolation: str = 'bicubic',
+    antialias: bool = True,
+) -> torch.Tensor:
+    """Resize absolute position embeddings to a target resolution via interpolation
+
+    Borrowed from https://github.com/bwconrad/flexivit.
```

Comment:
Contains a lot of code from FlexiViT and Aurora, but both are MIT licensed, so this doesn't change the license of the file as a whole.